tests/test_stp_global.rb

```diff
@@ -0,0 +1,533 @@
+                 @global.pathcost)
+  end
+
+  def test_get_set_vlan_forward_time
+    @global = create_stp_global
+    bddp = [['1-4,6,8-12', '10'], %w(14 8)]
+    @global.vlan_forward_time = bddp
+    assert_equal(bddp, @global.vlan_forward_time)
+    bddp = [['1-42', '19'], ['83-92,1000-2300', '13'],
+            ['3000-3700', '16']]
+    @global.vlan_forward_time = bddp
+    assert_equal(bddp, @global.vlan_forward_time)
+    @global.vlan_forward_time = @global.default_vlan_forward_time
+    assert_equal(@global.default_vlan_forward_time,
+                 @global.vlan_forward_time)
+  end
+
+  def test_get_set_vlan_hello_time
+    @global = create_stp_global
+    bddp = [['1-4,6,8-12', '5'], %w(14 8)]
+    @global.vlan_hello_time = bddp
+    assert_equal(bddp, @global.vlan_hello_time)
+    bddp = [['1-42', '10'], ['83-92,1000-2300', '6'],
+            ['3000-3700', '3']]
+    @global.vlan_hello_time = bddp
+    assert_equal(bddp, @global.vlan_hello_time)
+    @global.vlan_hello_time = @global.default_vlan_hello_time
+    assert_equal(@global.default_vlan_hello_time,
+                 @global.vlan_hello_time)
+  end
+
+  def test_get_set_vlan_max_age
+    @global = create_stp_global
+    bddp = [['1-4,6,8-12', '40'], %w(14 35)]
+    @global.vlan_max_age = bddp
+    assert_equal(bddp, @global.vlan_max_age)
+    bddp = [['1-42', '21'], ['83-92,1000-2300', '13'],
+            ['3000-3700', '16']]
+    @global.vlan_max_age = bddp
+    assert_equal(bddp, @global.vlan_max_age)
+    @global.vlan_max_age = @global.default_vlan_max_age
+    assert_equal(@global.default_vlan_max_age,
+                 @global.vlan_max_age)
+  end
+
+  def test_get_set_vlan_priorities
+    @global = create_stp_global
+    bddp = [['1-4,6,8-12', '4096'], %w(14 8192)]
+    @global.vlan_priority = bddp
+    @global.vlan_root_priority = bddp
+    @global.vlan_designated_priority = bddp
+    assert_equal(bddp, @global.vlan_priority)
+    assert_equal(bddp, @global.vlan_root_priority)
+    assert_equal(bddp, @global.vlan_designated_priority)
+    bddp = [['1-42', '40960'], ['83-92,1000-2300', '53248'],
+            ['3000-3700', '16384']]
+    @global.vlan_priority = bddp
+    @global.vlan_root_priority = bddp
+    @global.vlan_designated_priority = bddp
+    assert_equal(bddp, @global.vlan_priority)
+    assert_equal(bddp, @global.vlan_root_priority)
+    assert_equal(bddp, @global.vlan_designated_priority)
+    @global.vlan_priority = @global.default_vlan_priority
+    @global.vlan_root_priority = @global.default_vlan_root_priority
+    @global.vlan_designated_priority = @global.default_vlan_designated_priority
+    assert_equal(@global.default_vlan_priority,
+                 @global.vlan_priority)
+    assert_equal(@global.default_vlan_root_priority,
+                 @global.vlan_root_priority)
+    assert_equal(@global.default_vlan_designated_priority,
+                 @global.vlan_designated_priority)
+  end
+
+  def test_interface_stp_bpdufilter_change
```

Comment:
We have found for interface testing that it's best to make sure the interface is starting in a clean state.  You can use the following design pattern for this.

https://github.com/cisco/cisco-network-node-utils/blob/dae5fe60e6a72db50ab14dc79242f0f2b72e3c1e/tests/test_interface_channel_group.rb#L20